devnet-sdk/devstack/sysgo/sequencer.go

```diff
@@ -0,0 +1,218 @@
+	sys.T().Require().NoError(err)
+	sys.T().Cleanup(sqClient.Close)
+
+	l2sequencersRpcs := make(map[eth.ChainID]client.RPC)
+	for chainID, seqID := range s.l2sequencers {
+		seqRpc, err := client.NewRPC(sys.T().Ctx(), tlog, s.userRPC+"/sequencers/"+seqID.String(), opts...)
+		sys.T().Require().NoError(err)
+		sys.T().Cleanup(seqRpc.Close)
+
+		l2sequencersRpcs[chainID] = seqRpc
+	}
+
+	sys.AddSequencer(shim.NewSequencer(shim.SequencerConfig{
+		CommonConfig:       shim.NewCommonConfig(sys.T()),
+		ID:                 s.id,
+		Client:             sqClient,
+		L2SequencerClients: l2sequencersRpcs,
+	}))
+}
+
+func WithSequencer(sequencerID stack.SequencerID, l2CLID stack.L2CLNodeID, l1ELID stack.L1ELNodeID, l2ELID stack.L2ELNodeID) stack.Option {
+	return func(o stack.Orchestrator) {
+		orch := o.(*Orchestrator)
+		require := orch.P().Require()
+
+		logger := o.P().Logger().New("service", "op-test-sequencer", "id", sequencerID)
+
+		l1EL, ok := orch.l1ELs.Get(l1ELID)
+		require.True(ok, "l1 EL node required")
+
+		l2EL, ok := orch.l2ELs.Get(l2ELID)
+		require.True(ok, "l2 EL node required")
+
+		l2CL, ok := orch.l2CLs.Get(l2CLID)
+		require.True(ok, "l2 CL node required")
+
+		builderID := seqtypes.BuilderID("test-standard-builder")
+		committerID := seqtypes.CommitterID("test-standard-committer")
+		signerID := seqtypes.SignerID("test-local-signer")
+		publisherID := seqtypes.PublisherID("test-standard-publisher")
+
+		p2pKey, err := orch.keys.Secret(devkeys.SequencerP2PRole.Key(l2CLID.ChainID.ToBig()))
+		require.NoError(err, "need p2p key for sequencer")
+		raw := hexutil.Bytes(crypto.FromECDSA(p2pKey))
+
+		l2SequencerID := seqtypes.SequencerID(fmt.Sprintf("test-seq-%s", l2CLID.ChainID))
+
+		v := &config.Ensemble{
+			Endpoints: nil,
+			Builders: map[seqtypes.BuilderID]*config.BuilderEntry{
+				"test-standard-builder": {
+					Standard: &standardbuilder.Config{
+						L1EL: endpoint.MustRPC{
+							Value: endpoint.HttpURL(l1EL.userRPC),
+						},
+						L2EL: endpoint.MustRPC{
+							Value: endpoint.HttpURL(l2EL.userRPC),
+						},
+						L2CL: endpoint.MustRPC{
+							Value: endpoint.HttpURL(l2CL.userRPC),
+						},
+					},
+				},
+			},
+			Signers: map[seqtypes.SignerID]*config.SignerEntry{
+				"test-local-signer": {
+					LocalKey: &localkey.Config{
+						RawKey:  &raw,
+						ChainID: l2CLID.ChainID,
+					},
+				},
+			},
+			Committers: map[seqtypes.CommitterID]*config.CommitterEntry{
+				"test-standard-committer": {
+					Standard: &standardcommitter.Config{
+						RPC: endpoint.MustRPC{
+							Value: endpoint.HttpURL(l2CL.userRPC),
+						},
+					},
+				},
+			},
+			Publishers: map[seqtypes.PublisherID]*config.PublisherEntry{
+				"test-standard-publisher": {
+					Standard: &standardpublisher.Config{
+						RPC: endpoint.MustRPC{
+							Value: endpoint.HttpURL(l2CL.userRPC),
+						},
+					},
+				},
+			},
+			Sequencers: map[seqtypes.SequencerID]*config.SequencerEntry{
```

Comment:
At the moment the `ensemble` contains only a single concrete L2 sequencer, in the future we could setup more than one.